libcxx/test/benchmarks/algorithms/modifying/unique.bench.cpp

```diff
@@ -0,0 +1,131 @@
+  // state of the container after the operation.
+  {
+    auto bm = []<class Container>(std::string name, auto unique) {
+      benchmark::RegisterBenchmark(
+          name,
+          [unique](auto& st) {
+            std::size_t const size          = st.range(0);
+            constexpr std::size_t BatchSize = 10;
+            using ValueType                 = typename Container::value_type;
+            Container c[BatchSize];
+            ValueType x = Generate<ValueType>::random();
+            ValueType y = random_different_from({x});
+            for (std::size_t i = 0; i != BatchSize; ++i) {
+              c[i]      = Container(size);
+              auto half = size / 2;
+              std::fill_n(std::fill_n(c[i].begin(), half, x), half, y);
+            }
+
+            while (st.KeepRunningBatch(BatchSize)) {
+              for (std::size_t i = 0; i != BatchSize; ++i) {
+                benchmark::DoNotOptimize(c[i]);
+                auto result = unique(c[i].begin(), c[i].end());
+                benchmark::DoNotOptimize(result);
+              }
+
+              st.PauseTiming();
+              for (std::size_t i = 0; i != BatchSize; ++i) {
+                auto half = size / 2;
+                std::fill_n(std::fill_n(c[i].begin(), half, x), half, y);
+              }
+              st.ResumeTiming();
+            }
+          })
+          ->Arg(32)
+          ->Arg(1024)
+          ->Arg(8192);
+    };
+    bm.operator()<std::vector<int>>("std::unique(vector<int>) (contiguous)", std_unique);
+    bm.operator()<std::deque<int>>("std::unique(deque<int>) (contiguous)", std_unique);
+    bm.operator()<std::list<int>>("std::unique(list<int>) (contiguous)", std_unique);
+    bm.operator()<std::vector<int>>("rng::unique(vector<int>) (contiguous)", std::ranges::unique);
+    bm.operator()<std::deque<int>>("rng::unique(deque<int>) (contiguous)", std::ranges::unique);
+    bm.operator()<std::list<int>>("rng::unique(list<int>) (contiguous)", std::ranges::unique);
+  }
+
+  // Create a sequence of the form xxyyxxyyxxyyxxyyxxyy and unique
+  // adjacent equal elements.
+  //
+  // We perform this benchmark in a batch because we need to restore the
+  // state of the container after the operation.
+  {
+    auto bm = []<class Container>(std::string name, auto unique) {
+      benchmark::RegisterBenchmark(
+          name,
+          [unique](auto& st) {
+            std::size_t const size          = st.range(0);
+            constexpr std::size_t BatchSize = 10;
+            using ValueType                 = typename Container::value_type;
+            Container c[BatchSize];
+            ValueType x    = Generate<ValueType>::random();
+            ValueType y    = random_different_from({x});
+            auto alternate = [&](auto out, auto n) {
+              for (std::size_t i = 0; i != n; i += 2) {
+                *out++ = (i % 4 == 0 ? x : y);
+                *out++ = (i % 4 == 0 ? x : y);
```

Comment:
I think this would be a lot simpler if we go over the range in strides of four and do
```
*out++ = x;
*out++ = x;
*out++ = y;
*out++ = y;
```

And maybe add an assert that `size` is a multiple of four.